editor/src/components/canvas/floating-panels-state.tsx

```diff
@@ -0,0 +1,639 @@
+      (leftColumn.length === 0 || (draggedPanel === 'leftMenu1' && leftColumn.length === 1))
+    ) {
+      // dragging to the right of the left panel 2
+      return 'leftMenu1'
+    }
+
+    if (
+      dropTargetPanel === 'rightMenu1' &&
+      rightColumn.length > 0 &&
+      (rightColumn2.length === 0 || (draggedPanel === 'rightMenu2' && rightColumn2.length === 1))
+    ) {
+      // dragging to the left of the right panel 1
+      return 'rightMenu2'
+    }
+
+    if (
+      dropTargetPanel === 'rightMenu2' &&
+      rightColumn2.length > 0 &&
+      (rightColumn.length === 0 || (draggedPanel === 'rightMenu1' && rightColumn.length === 1))
+    ) {
+      // dragging to far right when something is already there
+      return 'rightMenu1'
+    }
+  }
+  return null
+}
+
+export function findDropAreaBeforeAfterColumn(
+  newPosition: WindowPoint,
+  panelState: PanelState,
+  canvasSize: Size,
+  draggedPanel: PanelName,
+): { newPanel: PanelName; switchWithPanel: PanelName | null } | null {
+  const EdgeDropAreaWidth = 60
+  const leftColumn = panelState.panelContent['leftMenu1'] ?? []
+  const leftColumn2 = panelState.panelContent['leftMenu2'] ?? []
+  const rightColumn = panelState.panelContent['rightMenu1'] ?? []
+  const rightColumn2 = panelState.panelContent['rightMenu2'] ?? []
+
+  const leftColumnFrame =
+    boundingRectangleArray(leftColumn.map((data) => data.frame)) ?? zeroWindowRect
+  const leftColumn2Frame =
+    boundingRectangleArray(leftColumn2.map((data) => data.frame)) ?? zeroWindowRect
+  const rightColumnFrame =
+    boundingRectangleArray(rightColumn.map((data) => data.frame)) ??
+    windowRectangle({ x: canvasSize.width, y: 0, width: 0, height: canvasSize.height })
+  const rightColumn2Frame =
+    boundingRectangleArray(rightColumn2.map((data) => data.frame)) ??
+    windowRectangle({ x: canvasSize.width, y: 0, width: 0, height: canvasSize.height })
+
+  const outerDropAreas: Array<{
+    targetPanel: PanelName
+    frame: WindowRectangle
+  }> = [
+    // between the left edge and the first left menu
+    {
+      targetPanel: 'leftMenu1',
+      frame: windowRectangle({
+        x: 0,
+        y: 0,
+        width: EdgeDropAreaWidth, //GapBetweenPanels,
+        height: canvasSize.height,
+      }),
+    },
+    // right side of the left menu
+    {
+      targetPanel: 'leftMenu2',
+      frame: windowRectangle({
+        x: leftColumnFrame.x + leftColumnFrame.width,
+        y: 0,
+        width: 80,
+        height: canvasSize.height,
+      }),
+    },
+    // left side of the right menu
+    {
+      targetPanel: 'rightMenu1',
+      frame: windowRectangle({
+        x: rightColumnFrame.x - 80,
+        y: 0,
+        width: 80,
+        height: canvasSize.height,
+      }),
+    },
+    // between the right edge and the last right menu
+    {
+      targetPanel: 'rightMenu2',
+      frame: windowRectangle({
+        x: rightColumn2Frame.x + rightColumn2Frame.width - EdgeDropAreaWidth, //GapBetweenPanels,
```

Comment:
is `GapBetweenPanels` still needed?

Reply:
Comment from pre-existing code, leaving it for now.